### PR TITLE
Support emcc 'upstream' branch

### PR DESCRIPTION
**Pull request by Beuc**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -17,7 +17,7 @@
 #    make install prefix=$HOME
 
 # -s WASM=0 -s SINGLE_FILE=1: allow loading zee.js synchronously
-EMCC=emcc -O2 -s WASM=0 -s SINGLE_FILE=1 -s EXPORTED_FUNCTIONS="['_gzopen', '_gzread', '_gzwrite', '_gzclose', '_gzerror', '_gzeof']" --pre-js pre.js --post-js post.js
+EMCC=emcc -O2 -s WASM=0 -s WASM_ASYNC_COMPILATION=0 -s SINGLE_FILE=1 -s EXPORTED_FUNCTIONS="['_gzopen', '_gzread', '_gzwrite', '_gzclose', '_gzerror', '_gzeof']" --pre-js pre.js --post-js post.js
```
Comment (kripken) [inline Makefile:20]: is the output file small enough to work in chrome?

Comment (Beuc) [reply Makefile:20]: Isn't WASM=0 meant to deal with this?

Comment (kripken) [reply Makefile:20]: Oh, I missed that. Yeah, without wasm there isn't an issue with code size. However, then we can remove `WASM_ASYNC_COMPILATION` as it has no effect.

Comment (Beuc) [reply Makefile:20]: `nodejs test.js` strongly disagrees :)
`Assertion failed: you need to wait for the runtime to be ready (e.g. wait for main() to be called)
`

Comment (kripken) [reply Makefile:20]: Ah, yes, I forgot - in upstream, wasm=0 tracks the wasm=1 path very closely, to make it easier to swap between the two. In particular, it looks like wasm all through the instantiation, all using a fake WebAssembly object. And that includes async startup. So yes, it is necessary there. Sorry for the mixup!

Comment (Beuc) [reply Makefile:20]: Thanks -- it's somewhat reassuring to see I'm not the only one struggling to keep all this in mind ;)

**Pull request merged**